### PR TITLE
[MNT] - Miscellaneous updates

### PR DESCRIPTION
**Pull request by TomDonoghue**
This is a set of miscellaneous updates (that stem for doing a sweep of usage in the T3 project). 

Updates include:
- adding and using a `check_param_type` helper function
- updating `plot_rate_by_time` to support a single 2d array that will be averaged
- add `plot_task_events` plot function
- adding some safety checking to `extract` functions

### CODE REVIEW

**Review by maestapereira — APPROVED** [review]
Looks good!

File: spiketools/utils/checks.py [hunk 1]
```diff
@@ -98,6 +98,35 @@ def check_param_lengths(params, labels, expected_length=None):
...
+    """Check that a parameter has an acceptable type.
+
+    Parameters
+    ----------
+    param : object
+        Parameter to check type of.
+    label : str
+        Label of the parameter being checked.
+    types : type or tuple of type
+        Type(s) to check the given paramter is one of.
```
Comment (claire98han) [inline spiketools/utils/checks.py:111]: Typo --> parameter

**Review by claire98han — APPROVED** [review]
Looks great!

File: spiketools/utils/checks.py (hunk 1 above)
Comment (TomDonoghue) [reply spiketools/utils/checks.py:111]: Good catch - thanks!

**Pull request merged**